nujo/utils/data/dataset.py

```diff
@@ -13,31 +13,35 @@ class Dataset:
 
     Parameters:
     -----------
-    name : will be downloaded from the UCI ML repo (for now)
+    - name : str, will be downloaded from the UCI ML repo (for now)
+    - type : str, type of dataset - either csv or image
+    - override : boolean, for loader, will override file if it exists
+    - download : boolean, should it be downloaded from uci repo
 
-    Returns:
-    --------
-    array : stores the csv dataset,
-    - floating point integers
-    - last column -> labels
 
     '''
-    def __init__(self, name: str):
-        self.name = HOME_DIR + name
-        if '.data' not in name:
-            self.name += '.data'
-        DatasetLoader(self)
+    def __init__(self,
+                 name: str,
+                 type: str,
+                 override: bool = True,
+                 download: bool = False):
+        self.name = name
+        self.type = type
+        loader = DatasetLoader(self.name, self.type, override)
+        if download is True:
+            loader.download()
+        self.X, self.y = loader.install()
 
     def __iter__(self):
         return DatasetIterator(self)
 
-    def __getitem__(self, position) -> array:
+    def __getitem__(self, position: Union[int, tuple]) -> ndarray:
```

Comment:
Use the `Tuple` class from `typing`. (something like `Union[int, Tuple[int, ...]`)